common/client-libs/validator-client/src/coconut/mod.rs

```diff
@@ -83,8 +90,19 @@ impl TryFrom<ContractVKShare> for EcashApiClient {
 
         let url_address = Url::parse(&share.announce_address)?;
 
+        // Note: we expect this to be used server-side (gateways / nodes / APIs) not client-side
```

Comment:
that one is tricky because clients will use it when attempting to get credentials locally, independently, but it kinda blurs the line of whether this dns is really needed in that instance and whether it really applies to the vpn

Reply:
Maybe the simplest solution for now is to revert this one. We have covered several longer lived clients with the other changes and I can't think of a clean solution for disentangling client vs non-client usage of this code right now.